### PR TITLE
Add detection of not running instances

### PR DESCRIPTION
**Pull request by m3rlinux**

### CODE REVIEW

**Comment by jherbel** [comment]
Hi, 
thank you very much for this PR. Except for bugfixes, we only merge PRs into the master branch (I am saying this because your PR is for the 1.6 branch). If you want, we can consider merging it into the master.
Best
Jörg

**Comment by m3rlinux** [comment]
Hi,
Thanks for your reply! I appreciate if you will merge the PR into the master branch.

File: checks/mongodb_instance [hunk 1]
```diff
@@ -35,7 +35,12 @@ def inventory_mongodb_instance(info):
 
 def check_mongodb_instance(_no_item, _no_params, info):
     infos = ["%s: %s" % (line[0].title(), line[1]) for line in info]
-    yield 0, ", ".join(infos)
+    for line in info:
+        status, messg = line
+        if status == "error":
+            yield 2, ", ".join(infos)
```
[the remark this answers is not in the document]
Comment (m3rlinux) [reply checks/mongodb_instance:41]: Yes! You're right! 
If error yield messg

[the remark this answers is not in the document]
Comment (jherbel) [reply checks/mongodb_instance:41]: Ok, then please update this PR accordingly

**Comment by m3rlinux** [comment]
```
<<<mongodb_instance:sep(9)>>>
error   Instance is down
```

File: checks/mongodb_instance [hunk 2]
```diff
@@ -35,7 +35,12 @@ def inventory_mongodb_instance(info):
 
 def check_mongodb_instance(_no_item, _no_params, info):
     infos = ["%s: %s" % (line[0].title(), line[1]) for line in info]
-    yield 0, ", ".join(infos)
+    for line in info:
+        status, messg = line
+        if status == "error":
+            yield 2, messg
+        else:
+            yield 0, ", ".join(infos)
```
Comment (jherbel) [inline checks/mongodb_instance:43]: Same as above, `yield 0,  mssg`, right?

**Review by jherbel — CHANGES_REQUESTED** [review]

File: checks/mongodb_instance (hunk 2 above)
Comment (m3rlinux) [reply checks/mongodb_instance:43]: nope, beacuase it's a list of list

Comment (jherbel) [reply checks/mongodb_instance:43]: You iterate over all lines in info, right? For each line, you check if there is an error. If there is an error, we get
`2, messg`
and I guess if not, we should get
`0, f"{status.title}: {messg}"`

**Review by jherbel — CHANGES_REQUESTED** [review]

File: checks/mongodb_instance (hunk 2 above)
Comment (jherbel) [reply checks/mongodb_instance:43]: Looks good, please change it to accordingly and I will merge it subsequently.

Comment (m3rlinux) [reply checks/mongodb_instance:43]: > PS: AFAIK String Interpolation is new in Python 3.6 but check_mk 1.6 use Python2

For this reason I forked only 1.6.0 branch.
So for the next times I guess if can I continue to work only on this branche (1.6.0) or not?

Comment (m3rlinux) [reply checks/mongodb_instance]: But the question is: How can I controbute on werks or patchs for the 1.6.0?

Comment (m3rlinux) [reply checks/mongodb_instance]: Thank Jörg for all your explainations and patience.
Just the last clarification. Do the approved changes end up in the 2.0.0 patches when working with the master?
Excuse my poor English

**Pull request closed without merging**